{% extends 'event/base.html' %}
{% load static %}

{%block content%}


<!DOCTYPE html>
<html>
  <head>
    <title>Help</title>
    <style>



.container {
  width: 100vw;
  height: 100vh;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  font-size: 1; /* Set font-size to 0 to remove white space between elements */
  display: grid;
  grid-template-columns:400px 600px 1fr;
  grid-template-rows: 50px 600px 100px;
  gap: 10px;
  padding: 10px;
  box-sizing: border-box;
  position: relative;
  outline: none;
}

.container div {
  padding: 10px;
  margin: 0px;
  z-index: 1;
}



.header {
  width: 1100px;
  grid-column-start: 1;
  grid-column-end: 4;
  outline: 2px;
  border: 1px solid #000000;
  background-color: rgb(240, 229, 198);
}

.gridcontainer{
  font-size: 0;
  width: 380px;
  height: 380px;
  position: relative;
  border: 1px solid #000000;
  padding: 0px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgb(240, 229, 198);
  
}

.content-large {
  grid-row-start: 2;
  grid-row-end: span 2;
  grid-column-start: 1;
  font-family: "IM Fell DW Pica","EB Garamond", serif;
  height: 700px;

  outline: 1px;
  border: 1px solid #000000;
  background-color: rgb(240, 229, 198);
}

.content-small {
  max-width: 500px;
  height: 600px;
  font-family: "IM Fell DW Pica","EB Garamond", serif;
  border: 1px solid #000000;
  padding: 10px;
  margin: 0px;
  z-index: 1;
  background-color: rgb(240, 229, 198);
}

.content-question_answer {
  width: 600px;
  height: 700px;
  font-family: "IM Fell DW Pica","EB Garamond", serif;
  border: 1px solid #000000;
  padding: 10px;
  margin: 0px;
  z-index: 1;
  background-color: rgb(240, 229, 198);
}

.footer {
  grid-column-start: 1;
  grid-column-end: span 3;
  grid-row-start: 3;
  border: 1px solid #000000;
  background-color: rgb(240, 229, 198);
}


</style>
  </head>
  <body>
    <div class="container">
     <div class="header">
      <a class="nav-link" href="{% url 'home' %}">Home</a>&nbsp;&nbsp;&nbsp;&nbsp;
      <a class="nav-link" href="{% url 'login' %}">Log in</a>&nbsp;&nbsp;&nbsp;&nbsp;
      <a class="nav-link" href="{% url 'managegenomes' %}">Manage genomes</a>&nbsp;&nbsp;&nbsp;&nbsp; 
        <a class="nav-link" href="{% url 'register_user' %}">Register</a>&nbsp;&nbsp;&nbsp;&nbsp; 
        <a class="nav-link" href="{% url 'logout' %}">Log out</a>&nbsp;&nbsp;&nbsp;&nbsp; 
        <a class="nav-link" href="{% url 'editSettings' %}"> Edit Settings</a>&nbsp;&nbsp;&nbsp;&nbsp;
        <a class="nav-link" href="{% url 'help' %}">Help</a>&nbsp;&nbsp;&nbsp;&nbsp;
      User: {{user.userprofile}}


     </div>
     
     <div class="content-large">
    Home tab:<br>
    <br>
    Your investigated genomes are shown here.<br>
    <br>
    Name: file name for the genome<br>
    Nickname: not in use<br>
    Genome directory: the location of the file<br>
    Genome size: size of the genome in nt<br>
    IS footprint: number of nt consisting of IS<br>
    <br>
    Buttons:<br>
    <br>
  
    Analyse: computes the size of the genome<br>
    <br>
    Prepare: Opens genome file in fasta or gb format,
    <br>&nbsp;&nbsp;&nbsp;&nbsp; concatenates all records and <br>
    &nbsp;&nbsp;&nbsp;&nbsp;writes a file _conc.fa to the workfiles directory<br><br>
    Blast: Blastx:s the nucleotide file against the transposase_protein_database<br>
    &nbsp;&nbsp;&nbsp;&nbsp;Creates blast_files_dir if neccessary.<br>
    &nbsp;&nbsp;&nbsp;&nbsp;Writes results to genomename + _blast1results.xml<br><br>
    Analyse Blast: Parses the xml results file, writes the found hits in a gb file<br>
    &nbsp;&nbsp;&nbsp;&nbsp; Creates blast_analysis_dir if this does not exist<br><br>
    Make Footprints: Blastx:s the footprints of ISs from the previous search,<br>
    &nbsp;&nbsp;&nbsp;&nbsp;as many are hits on the same region of the genome.<br>
    &nbsp;&nbsp;&nbsp;&nbsp;blastx:s the footprint to discriminate ISs.<br>
    &nbsp;&nbsp;&nbsp;&nbsp;stores files in analysed_gb_files_dir, with file ending "_curated.gb"<br><br>
    Analyse results: Writes a csv file with the found ISs, stored in is_list_csv_file_dir<br>
    &nbsp;&nbsp;&nbsp;&nbsp;Makes a picture of the distribution of IS fragments.<br>


    </div> <!--End contentlarge-->
    

    <!--Questions window-->
     <div class="content-question_answer">
      Manage genomes tab:<br>
      <br>
      This is where you add or remove genomes from an investigation<br>
      <br>
      By default, the genome directory is set to static/genomes,<br> 
      in the folder containing this program<br>
        <br>
      To change the genome directory:<br><br>
        Enter a directory with .gb files in "Manage genomes",under "Genomes in directory".
        <br>
        e.g. "c:\Users\Eris\Documents\visapp_proj\static\genomes"<br><br>
        Push "set directory"<br>
        <br>

      To add genomes to the list of investigated genomes:<br><br>

      Checkmark the files you want to transfer to "Genomes in database"<br>
      <br>
      Push "add to database"
      <br><br>
      <br>
      Other notes:<br>
      <br>
        The protein database must be an ncbi formatted protein database<br>
        this consists of three files, example:<br>
        is_aa_30_nov2016.fa.phr<br>
        is_aa_30_nov2016.fa.pin<br>
        is_aa_30_nov2016.fa.psq<br>
        <br>
        The path for these files would be entered as e.g.<br>
        C:/autothink/is_aa_30_nov2016.fa<br>
        "is_aa_30_nov2016.fa" was the name of the nr file fed into the ncbi script for making search databases<br>
        A copy of these files are included in visualAutothink\visapp_proj\isdatabase<br>
        <br>
    
       
   

        Results are stored in the is_list_csv_file_dir for a csv with info of identified ISs, and 
        analysed_gb_files_dir contains genbank files with additional information.
  
   
      <br>


     </div>

    


<script>

  function updatePage(x, y) {

    var user = '{{ user.userprofile.question.area1 }}';
    if (user == "utility") {

      var input_x = document.createElement('input');
      input_x.type = 'hidden';
      input_x.name = 'sent_x';
      input_x.value = x;

      var input_y = document.createElement('input');
      input_y.type = 'hidden';
      input_y.name = 'sent_y';
      input_y.value = y;

      var input_action = document.createElement('input');
      input_action.type = 'hidden';
      input_action.name = 'sent_action';
      input_action.value = 'answer';

      var form = document.getElementById('mapform');
      form.appendChild(input_x);
      form.appendChild(input_y);
      form.appendChild(input_action);
      form.submit();
    }
}

function receiveAnswer(command, answer,unused) {

  var command_input = document.createElement('input');
  command_input.type = 'hidden';
  command_input.name = 'command';
  command_input.value = 'answer';
  
  var answer_input = document.createElement('input');
  answer_input.type = 'hidden';
  answer_input.name = 'answer';
  answer_input.value = answer;
  //alert("Selected value: " + dropdownValue);
  //console.log('Dropdown value:', document.getElementById('dropdown').value);
  
  var form = document.getElementById('question-form');
  form.appendChild(command_input);
  form.appendChild(answer_input);
  form.submit();
}




  </script>


{%endblock%}